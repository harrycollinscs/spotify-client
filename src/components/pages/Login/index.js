import { connect } from 'react-redux';
import Login from '../../templates/Login';

const mapStateToProps = () => {

}

const mapDispatchToProps = () => {
  
}

export default connect()(Login);
