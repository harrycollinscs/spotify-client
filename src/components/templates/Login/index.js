import React from "react";
import { NavLink } from "react-router-dom";
import styled from "styled-components";
import { full_auth_url } from "../../../config";
import CTA from "../../atoms/CTA";
import Page from "../../atoms/Page";
import VinylImage from "../../../resources/images/vinyls.png";

const StyledButtonContainer = styled.div`
  align-items: center;
  display: flex;
  justify-content: space-evenly;
`;

const Container = styled.div`
  position: relative;
  align-items: center;
  display: flex;
  justify-content: space-evenly;
  margin: -30px -30px 30px -30px;
  margin-bottom: 30px;
`;

const StyledHeroText = styled.h1`
  position: absolute;
  top: 50%;
  color: white;
  font-weight: bold;
`;

const StyledImage = styled.img`
  width: 100vw;
  height: 40vh;
  opacity: 0.75;
`;

const ButtonAction = () => {
  window.location.href = full_auth_url;
};

const Login = () => (
  <Page withNavigation={false} darkBackground={true}>
    <Container>
      <StyledImage src={VinylImage} />
      <StyledHeroText>Hey there! Sign in to see more.</StyledHeroText>
    </Container>

    <StyledButtonContainer>
      <NavLink to={"/login"}>
        <CTA action={ButtonAction}>Log in with Spotify</CTA>
      </NavLink>
    </StyledButtonContainer>
  </Page>
);

export default Login;
