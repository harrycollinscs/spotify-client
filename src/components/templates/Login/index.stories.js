import React from "react";
import Login from "./index";

export default {
  title: "Templates/Login",
  component: Login,
};

const Template = (args) => <Login {...args} />;

export const Default = Template.bind({});
