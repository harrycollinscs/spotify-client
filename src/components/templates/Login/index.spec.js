import React from "react";
import Login from "./index";
import { shallow } from "enzyme";

describe("Templates/Login", () => {

  const component = <Login />;

  it("Should match snapshot", () => {
    const render = shallow(component);
    expect(render).toMatchSnapshot();
  });
});
