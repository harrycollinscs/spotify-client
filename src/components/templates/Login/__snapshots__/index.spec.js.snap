// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates/Login Should match snapshot 1`] = `
<Page
  darkBackground={true}
  withNavigation={false}
>
  <styled.div>
    <styled.img
      src={null}
    />
    <styled.h1>
      Hey there! Sign in to see more.
    </styled.h1>
  </styled.div>
  <styled.div>
    <NavLink
      to="/login"
    >
      <CTA
        action={[Function]}
      >
        Log in with Spotify
      </CTA>
    </NavLink>
  </styled.div>
</Page>
`;
